tracer/test/test-applications/integrations/Samples.DataStreams.RabbitMQ/Program.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Text;
+using RabbitMQ.Client;
+
+namespace Samples.DataStreams.RabbitMQ
+{
+    public static class Program
+    {
+        private static readonly string Queue = nameof(Queue);
+        private static readonly string Exchange = nameof(Exchange);
+        private static readonly string RoutingKey = nameof(RoutingKey);
+        private static readonly string Message = nameof(Message);
+        private static readonly string Host = Environment.GetEnvironmentVariable("RABBITMQ_HOST") ?? "localhost";
+        
+        public static void Main(string[] args)
+        {
+            var factory = new ConnectionFactory() { HostName = Host };
+            using var connection = factory.CreateConnection();
+            using var model = connection.CreateModel();
+        
+            // produce/consume operation (direct exchange)
+            PublishMessageToQueue(model, Queue, Message);
+            var msg = GetMessage(model, Queue);
+
+            // continuing the chain
```

Comment:
nit: I suggest making this an `async Task` Main method, and adding an `await Task.Yield()` to try to force testing the flow behaviour. 

You _potentially_ should also add a test that creates separate `Task`, because the async local will flow across the sync context, and may give you unexpected Parent context values. 

This is somewhat to be expected, but given the way various abstractions use RabbitMQ, I won't be that surprised if we end up getting incorrect pathways in customer applications :(. Some of that _may_ be unavoidable, but it's something to be aware of. 

Just to be concrete, if you have this 

```csharp
var factory = new ConnectionFactory() { HostName = Host };
using var connection = factory.CreateConnection();
using var model = connection.CreateModel();

// produce/consume operation (direct exchange)
PublishMessageToQueue(model, Queue, Message);
var msg = GetMessage(model, Queue);

// continuing the chain, this _will_ flow correctly
await Task.Run(() =>
{
     var queueName = PublishMessageToExchange(model, Exchange, RoutingKey, msg);
     GetMessage(model, queueName);
})

// But if you run it _again_, you theoretically would want this to
// follow on from the PublishMessageToExchange, but it won't. It
// will follow on from the PublishMessageToQueue
await Task.Run(() =>
{
     var queueName = PublishMessageToExchange(model, Exchange, RoutingKey, msg);
     GetMessage(model, queueName);
})
```